### PR TITLE
[API][Account] Password change feature

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         | master 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | related to #11962
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Resources/config/services/api.xml [hunk 1]
```diff
@@ -153,6 +153,11 @@
             <argument>shop</argument>
         </service>
 
+        <service id="sylius.behat.api_platform_client.shop.user" class="Sylius\Behat\Client\ApiPlatformClient" parent="sylius.behat.api_platform_client">
```
Comment (GSadee) [inline src/Sylius/Behat/Resources/config/services/api.xml:156]: Do you use this service somewhere?

File: src/Sylius/Bundle/ApiBundle/Command/ChangeShopUserPassword.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Command;
+
+/**
+ * @experimental
+ */
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Command/ChangeShopUserPassword.php:18]: Could be in one line 😃

File: src/Sylius/Bundle/ApiBundle/CommandHandler/ChangeShopUserPasswordHandler.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+/** @experimental */
+final class ChangeShopUserPasswordHandler implements MessageHandlerInterface
+{
+    /** @var PasswordUpdaterInterface */
+    private $passwordUpdater;
+
+    /** @var UserRepositoryInterface */
+    private $userRepository;
+
+    public function __construct(PasswordUpdaterInterface $passwordUpdater, UserRepositoryInterface $userRepository) {
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/ChangeShopUserPasswordHandler.php:32]:
```suggestion
    public function __construct(PasswordUpdaterInterface $passwordUpdater, UserRepositoryInterface $userRepository)
    {
```

File: src/Sylius/Bundle/ApiBundle/CommandHandler/ChangeShopUserPasswordHandler.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        $this->passwordUpdater = $passwordUpdater;
+        $this->userRepository = $userRepository;
+    }
+
+    public function __invoke(ChangeShopUserPassword $changeShopUserPassword){
+
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/ChangeShopUserPasswordHandler.php:38]:
```suggestion
    public function __invoke(ChangeShopUserPassword $changeShopUserPassword)
    {
```

File: src/Sylius/Bundle/ApiBundle/spec/DataTransformer/LoggedInShopUserIdAwareCommandDataTransformerSpec.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    function let(UserContextInterface $userContext): void
+    {
+        $this->beConstructedWith($userContext);
+    }
+
+    function it_supports_only_shop_user_id_commands(
+        ShopUserIdAwareInterface $shopUserIdAware,
+        CommandAwareDataTransformerInterface $commandAwareDataTransformer
+    ): void
+    {
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/DataTransformer/LoggedInShopUserIdAwareCommandDataTransformerSpec.php:38]:
```suggestion
    ): void {
```

**Review by GSadee — APPROVED** [review]

**Review by AdamKasp — APPROVED** [review]

**Comment by lchrusciel** [comment]
Closing in favor of #11998

**Pull request closed without merging**